### ISSUE TITLE
How to install Org from github:bzg/org-mode using `package!` ?

### ISSUE THREAD

**Issue by fuxialexander** [hlissner/doom-emacs#315]
Seems the following does not work.
```elisp
(package! org :recipe (:fetcher github :repo "bzg/org-mode" :files ("lisp/*.el" "contrib/lisp/*.el")))
```

**Comment by hlissner** [comment]
Because `org` already exists as a package built into Emacs, you'll need to use a different name for the package:

```emacs-lisp
(package! org-alt :recipe (:fetcher github :repo "bzg/org-mode"
```

However, be warned. When you try to do `(require 'org)`, Emacs will pick up the built-in version first!  So you'll need something like this in the config.el file of your private module:

```emacs-lisp
;; Ensure ELPA org is prioritized above built-in org.
(when-let* ((path (locate-library "org" nil doom--package-load-path)))
  (setq load-path (delete path load-path))
  (push (file-name-directory path) load-path))
```

**Comment by fuxialexander** [comment]
Thanks a lot! Finally got it work using the code below.

*packages.el*
```elisp
(package! org-plus-contrib :recipe (:fetcher github :repo "bzg/org-mode" :files
("lisp/*.el" "contrib/lisp/*.el")))
```
*config.el*
```elisp
(when-let* ((path (car (file-expand-wildcards "~/.emacs.d/.local/packag
es/elpa/org-plus-contrib*"))))
  (setq load-path (delete path load-path))
  (push (file-name-directory path) load-path))
```

**Comment by hlissner** [comment]
No problem.

Just curious, why are you using bsg/org-mode? `lang/org` installs org-plus-contrib from emacsmirror and I'm wondering if there's a difference between the two.

**Comment by fuxialexander** [comment]
It's several days or weeks ahead which is more bleeding edge. Recently I saw on twitter that a new [feature](https://github.com/bzg/org-mode/commit/8983ac917b1ca1b46697a5d27655e4ef71fa6537) was added to org-colview.el, and I'd like to try it out. 

BTW, seems the code above does not fully works. It seems that Emacs are blending some code from the bundled / ELPA org with the git one. Anyway for now I just clone the repository and add it to the user/init.el. Will try to resolve it later (maybe I need to modify the load-path earlier, like what I do now, rather than in org/config.el. Perhaps in org/init.el it will work).
